Dj remix branch

Snupkin opened the pull request:
lots of updates, check the commits messages.
in short: signup works
route guarding works
birthdate field added to signup
minor functionality tweaks
splash page during signin

testing: check route guarding (type urls that should be inaccessible. it should reroute to sign in page)
check new user creation, including entering erroneous credentials and missing credentials
check for bugs

testing: check favorites functionality - can add multiple favorites, cant add same one twice, button changes color/text, can remove favorites, adding/removing errors functionality

testing: check to see if AddFavorites can be deleted as i believe it is now obsolete and replaced by MyLibrary

Snupkin committed:
Update ArticleLibrary.js

Snupkin committed:
added firestore config in index.js

Snupkin committed:
Merge branch 'dj-remix-branch' of https://github.com/cocassel/atlas-life-systems into dj-remix-branch

Snupkin committed:
Merge branch 'celeste-branch' of https://github.com/cocassel/atlas-life-systems into celeste-branch

Snupkin committed:
tested route guarding

currently not functional - articles wont show up on dashboard. going to merge celeste branch into dj remix branch and test again

Snupkin committed:
Merge branch 'celeste-branch' into dj-remix-branch

Snupkin committed:
Route guarding

Implementation of Route guarding for:
Signup/Signin when logged in
quiz,/add favourite/article library when logged out

Snupkin committed:
minor splash screen changes

Snupkin committed:
Merge branch 'master' into dj-remix-branch

Snupkin committed:
Signup form updates

can now sign up a user from app
date of birth field updated
made fields required

Snupkin committed:
summary/content rename and articledetails route guard fix

stefan changed the content proprerty of articles to summary
i had a previously unknown bug in articledetails where {auth }=this.props needed to be changed to {auth} = props

Snupkin committed:
initial icon update

pink initial icon now shows the users initials instead of just dj

Snupkin committed:
favourite renaming

renamed some files and functions from 'favourite' to 'favorite'
renamed 'favouriteLibrary' to 'MyLibrary'
updated the imports

cocassel commented:
Everything looks great! Two minor comments: 
1. When I tried out the route guarding, it did bring me to the login page but the automatically-filled fields overlapped with the field labels (see image below)
<img width="224" alt="Screen Shot 2020-08-21 at 3 48 30 PM" src="https://user-images.githubusercontent.com/41701024/90928998-e5b30a80-e3c5-11ea-85bc-3bd2da0fbca4.png">
When I click on one of the fields, it fixes itself (see below)
<img width="278" alt="Screen Shot 2020-08-21 at 3 48 35 PM" src="https://user-images.githubusercontent.com/41701024/90929056-024f4280-e3c6-11ea-86f3-3d3a4fc8af68.png">
2. Love the little "Please fill out this field" pop-up when you leave an empty field in sign-up. Do you think we should do that for the quiz too instead of just greying out the submit button? I think we should be consistent so let me know what you think!

Snupkin committed:
Favorites Section Update

addFavorite function now adds article to users collection myLibrary field.
shows favorites on My-Library page.
currently only working for 1 article at a time

Snupkin committed:
Favorites

Full functionality should be in place
